### ISSUE TITLE
Copy paste with formating

### ISSUE THREAD

**Issue by virgile-dev** [numerique-gouv/impress#436]
## Feature Request

**Is your feature request related to a problem or unsupported use case? Please describe.**
Users don't want to have to redo formating when they copy paste from another source.

**Describe the solution you'd like**
I'd like blocknote to be able to recognize formating from Docx and Gdoc to start. 
Recognize = create blocks for each paragraph or title, recognize title level, recognize text formating (bold, italic etc.), tables etc.

**Comment by YousefED** [comment]
fyi, this was partially [fixed](https://github.com/TypeCellOS/BlockNote/pull/1224) in the latest version of BlockNote. Compare pasting from docx here:
- https://blocknote-main.vercel.app/basic/minimal (most recent version of BlockNote)
- vs. in la suite / docs

I suppose this might already change the priority. However, we definitely don't support all formatting yet.

Of course this can be an "endless" task until we add support for all formatting options that a Word / Gdocs document has - but this is intentionally not the scope of Docs / BlockNote. My suggestion to scope this issue would be to add support for copying back the document that's generated in the [Converting to Docx example](https://blocknote-main.vercel.app/interoperability/converting-blocks-to-docx), wdyt?

**Comment by virgile-dev** [comment]
@YousefED Could we improve markdown handling when copy pasted into blocknote ?
Right now it doesn't it creates a block for each line but it doesn't do the automatic conversion into titles etc.

**Comment by YousefED** [comment]
Yes, atm we don't parse markdown when pasting plain text. We can add this, only thing is that I'm not sure if this will cause issues for when a user just wants to paste plain text (without parsing markdown). Not sure how often that happens, but in that case we need a UX for this.

We could try just implementing markdown paste and make it the default -> then see if we get any "complaints"

**Comment by YousefED** [comment]
@virgile-dev this has been added in BlockNote: https://github.com/TypeCellOS/BlockNote/pull/1490
